### DISCUSSION TITLE
Auth0 Lock Library Support

### DISCUSSION THREAD

**Discussion by ankur-hyperspace** [nextauthjs/next-auth#2435] [Help]
### Question 💬

Can I use Auth0Lock?
Auth0Lock provides to add a custom field so how can use Auth0Lock.

https://auth0.com/docs/libraries/lock

### How to reproduce ☕️

```
{
additionalSignUpFields: [{
    name: "address",
    placeholder: "enter your address",
    // The following properties are optional
    icon: "https://example.com/assests/address_icon.png",
    prefill: "street 123",
    validator: function(address) {
      return {
         valid: address.length >= 10,
         hint: "Must have 10 or more chars" // optional
      };
    }
  },
  {
    name: "full_name",
    placeholder: "Enter your full name"
  }]
}
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
We already support Auth0, see: https://next-auth.js.org/providers/auth0

I'm not sure what Auth0 Lock would bring to the table in addition. After a very quick look, it seems that you would need to ember a script in your frontend, which differs from any of our other providers, so I don't think this would be a straight forward addition, if necessary at all. I would like to hear more about the reasons you would like to use this over our existing Auth0 integration?
